### ISSUE TITLE
Allow disabling plugins

### ISSUE THREAD

**Issue by jayrhynas** [supermarin/Alcatraz#376]
It would be handy to have a way to quickly disable a plugin without completely removing it. This is handy for toggling plugins you only use occasionally, or for troubleshooting. Similar to how Chrome extensions can be disabled.

This can be achieved by moving plugins to a different folder, such as "~/Library/Application Support/Developer/Shared/Xcode/Plug-ins (Disabled)".

**Comment by supermarin** [comment]
@jayrhynas what would be advantage over just removing them? installation is usually super quick

**Comment by jayrhynas** [comment]
I agree, installation is quick, but some disabling is still useful for when you want to temporarily disable some or all plugins, and you either don't have internet access (to reinstall) or you don't want to write down the list of plugins so you can remember which ones to install.

I admit though this feature is just a minor convenience, and shouldn't be a priority. You can still manually "disable" plugins by moving them out of the Plug-ins folder.

**Comment by jayrhynas** [comment]
Also, I can do the work to contribute this myself, but I wanted to see if there's actually any interest before tackling it.

**Comment by jurre** [comment]
I'd say it's probably not a common enough use case to add code for in Alcatraz, but if there are many people that would find this useful, we could consider it I suppose.

**Comment by ForkPanda** [comment]
[Portal Here! A script for reload all plugin-ins](https://github.com/ForkPanda/RescueXcodePlug-ins)

**Comment by revolter** [comment]
This is useful if you have a fork of a plugin installed or even a plugin in development. Installing it retrieves the "original" plugin, while (re)enabling it retains the current version (and build) of the plugin, since it's the exact same file. So, yes, I "upvote" this idea.

**Comment by Cueball** [comment]
Just came here looking exactly for this feature. (Implicit +1 on this if it wasn't obvious...)

The key factor is that disabling one/all plugins should not require you to remember which plugins you've had installed - you can disable them and re-enable them as a simple troubleshooting technique. (And it may or may not integrate nicely with #413 which I found earlier.)

Specific use case: I've just been given a .playground that's not presenting properly in Xcode. I want to disable all plugins to make sure that no 3rd-party plugin is causing the problem. Easiest first step would be to disable all plugins and restart Xcode. But instead I have to document all my plugins, uninstall them, relaunch Xcode, and troubleshoot the .playground (which could just as easily be .xcodeproj). To get my plugins back, I have to search for each plugin and manually re-enable them one-by-one - that's gonna get old really fast.

TLDR: need disable/enable per-plugin and en masse for troubleshooting plugins.

**Comment by revolter** [comment]
Well, there is an (far) easier solution than yours, @Cueball, thought it's not the easiest, and I still consider this useful. Go to `/Users/User/Library/Application Support/Developer/Shared/Xcode/Plug-ins` and move all the plugins somewhere else (I usually just move them outside their parent folder, as siblings with the `Plug-ins` folder) and move them back when you're done.

**Comment by Matthewspear** [comment]
+1

Even though the folder based solution is easy, it's the type of feature that comes in handy when troubleshoot and trying to work out exactly which plugin might be causing the problem! 

For troubleshooting on a per plugin basis you have to re-introduce plugin by plugin, where ticking an enable / disable tickbox would be alot easer and more accessible than either ```mv ~/...``` or dragging and dropping.

**Comment by jurre** [comment]
Ok it seems like more people than I thought would like this.

Does anyone has any UI suggestions? And *if* we would decide to add this, would anyone be willing to put in the work?

**Comment by Matthewspear** [comment]
UI wise - I imagined just having either a tick box:
<img width="537" alt="screen shot 2016-01-14 at 8 14 33 pm" src="https://cloud.githubusercontent.com/assets/1668863/12336265/904ad340-bafb-11e5-9678-6dc0f7351df3.png">
or an on off switch:
<img width="506" alt="screen shot 2016-01-14 at 8 13 27 pm" src="https://cloud.githubusercontent.com/assets/1668863/12336263/8c6ac028-bafb-11e5-9782-97a26b6ef20d.png">

I am by no means a designer but this is roughly what I was thinking, obviously the text could change when selected (Enable > Enabled in first case, Enable > Disable in second).

Work wise - I'm a student just about to start exam season, so not this month also I'm a Swift developer with only a little bit of Obj-C knowledge but I'd be willing to try!

**Comment by sebastienboisvert** [comment]
Something like holding down a specific key combo (maybe customizable?) during Xcode launch would be handy.

**Comment by Petter-W** [comment]
Is this still considered?
